### PR TITLE
refactor: update drawer logic and container layout

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -**
Updates Gradebook "Drawer" layout and "container" workflow to be more sensible.
The main goal of this PR is to clean up the Drawer component in order to get it tested.  However, the Drawer component as it existed was performing a weird set of end-runs in order to accomplish its goal, that could be much more easily accomplished with a bit of re-organization.

summary in ticket:
JIRA: [AU-8](https://openedx.atlassian.net/browse/AU-8)

[Diff of Drawer/index.jsx after becoming WithSidebar/index.jsx] (https://gist.github.com/muselesscreator/0da092234578fcf05386d3d24727fce3)
[Diff of Gradebook/index.jsx after becoming GradesTab/index.jsx] (https://gist.github.com/muselesscreator/9ef258522a0bf88cc0c29fd10b868a56)

**What changed?**

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by leangseu-edx — APPROVED** [review]
LGTM

**Review by mattcarter — APPROVED** [review]
that's a lot of file touches.  Tests pass, functionality works.  Approved 👍

**Review by jansenk — APPROVED** [review]
Looks okay to me!

File: src/data/thunkActions/app.js [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
+};
+
+export const filterMenu = StrictDict({
+  close: () => (dispatch, getState) => {
+    if (selectors.app.filterMenu.open(getState())) {
+      dispatch(module.filterMenu.toggle());
+    }
+  },
+  handleTransitionEnd: (event) => (dispatch) => {
+    if (event.currentTarget === event.target) {
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/data/thunkActions/app.js:23]: honestly not 100% sure.  Some version of checking if the transition is done.  maybe whether the drawer is still open?  Is copying existing implementation

File: src/data/thunkActions/app.js [hunk 2]
```diff
@@ -1,8 +1,38 @@
...
+      dispatch(actions.app.filterMenu.endTransition);
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/data/thunkActions/app.js:24]: yas!

**Pull request merged**